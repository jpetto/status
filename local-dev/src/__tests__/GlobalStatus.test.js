import React from 'react';
import renderer from 'react-test-renderer';

import GlobalStatus from '../components/GlobalStatus';

test('healthy global status', () => {
    const component = renderer.create(
        <GlobalStatus status='healthy'/>
    );

    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
});

test('warning global status', () => {
    const component = renderer.create(
        <GlobalStatus status='warning'/>
    );

    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
});

test('failed global status', () => {
    const component = renderer.create(
        <GlobalStatus status='failed'/>
    );

    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
});

test('pending global status', () => {
    const component = renderer.create(
        <GlobalStatus status='pending'/>
    );

    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
});

test('sends a desktop notification', () => {
    // tests can't actually send a desktop notification
    GlobalStatus.prototype.sendDesktopNotification = jest.fn(() => true);
    // tests don't like looking for file paths, so we create a predictable stub
    GlobalStatus.prototype.statusToIcon = jest.fn(() => 'stub');

    const component = renderer.create(
        <GlobalStatus
            status='healthy'
            desktopNotify={true}
            notifyMessage={true}
            message='playtime is fun'/>
    );

    expect(GlobalStatus.prototype.sendDesktopNotification).toHaveBeenCalledWith('playtime is fun', 'stub');
});
