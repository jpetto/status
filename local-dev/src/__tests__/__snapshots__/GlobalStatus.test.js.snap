exports[`test failed global status 1`] = `
<div
  className="alert alert-danger"
  role="alert" />
`;

exports[`test healthy global status 1`] = `
<div
  className="alert alert-success"
  role="alert" />
`;

exports[`test pending global status 1`] = `
<div
  className="alert alert-info"
  role="alert" />
`;

exports[`test warning global status 1`] = `
<div
  className="alert alert-warning"
  role="alert" />
`;
